packages/sdk/src/utils.ts

```diff
@@ -182,8 +182,8 @@ const generateShareFilename = /* @__PURE__ */ (
 
 const getResourceUrl = (module: ModuleInfo, sourceUrl: string): string => {
   if ('getPublicPath' in module) {
-    const publicPath = new Function(module.getPublicPath)();
-    return `${publicPath}${sourceUrl}`;
+    const publicPath = new Function('return ' + module.getPublicPath)();
```

Comment:
Allowing users to specify fully functions instead of expressions will give more power.